### ISSUE TITLE
Management image can't be successfully run

### ISSUE THREAD

**Issue by visibletrap** [docker-library/rabbitmq#2]
Executed the command `$ docker run rabbitmq:3-management` gives this error

```
=ERROR REPORT==== 6-Jan-2015::08:54:49 ===
** Generic server rabbit_mgmt_external_stats terminating 
** Last message in was emit_update
** When Server state == {state,1048576}
** Reason for termination == 
** {badarg,[{erlang,list_to_binary,[undefined],[]},
            {rabbit_mgmt_external_stats,log_location,1,[]},
            {rabbit_mgmt_external_stats,'-infos/2-lc$^0/1-0-',2,[]},
            {rabbit_mgmt_external_stats,'-infos/2-lc$^0/1-0-',2,[]},
            {rabbit_mgmt_external_stats,emit_update,1,[]},
            {rabbit_mgmt_external_stats,handle_info,2,[]},
            {gen_server,handle_msg,5,[{file,"gen_server.erl"},{line,607}]},
            {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,227}]}]}
```

**Comment by tianon** [comment]
Ack, what.  I'm positive that was working when I tested it originally, but now it's not (ie, I can confirm). :disappointed:

**Comment by tianon** [comment]
Ok, so the management GUI does come up, which is what made me think this was working.  This error must be for some side process not starting.

Any ideas? :smile:

**Comment by visibletrap** [comment]
The management GUI didn't come up for me when I was creating this issue, but it does come up now. Not really sure what was happening.

**Comment by tianon** [comment]
I'm going to reopen this because there's definitely something wrong going on here and we need to figure out what it is. :cry:

**Comment by themasterchef** [comment]
I too get the same error, but for me:

- The GUI never comes up
- The message below is logged about twice a second repeatedly
- After about 10 seconds of error logging the whole image crashes for me

```
=ERROR REPORT==== 28-Jan-2015::16:21:02 ===
** Generic server rabbit_mgmt_external_stats terminating 
** Last message in was emit_update
** When Server state == {state,1048576}
** Reason for termination == 
** {badarg,[{erlang,list_to_binary,[undefined],[]},
            {rabbit_mgmt_external_stats,log_location,1,[]},
            {rabbit_mgmt_external_stats,'-infos/2-lc$^0/1-0-',2,[]},
            {rabbit_mgmt_external_stats,'-infos/2-lc$^0/1-0-',2,[]},
            {rabbit_mgmt_external_stats,emit_update,1,[]},
            {rabbit_mgmt_external_stats,handle_info,2,[]},
            {gen_server,handle_msg,5,[{file,"gen_server.erl"},{line,607}]},
            {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,227}]}]}

=INFO REPORT==== 28-Jan-2015::16:21:02 ===
    application: rabbitmq_management_agent
    exited: shutdown
    type: temporary
```

**Comment by sitamet** [comment]
My image also crashes.
```
=INFO REPORT==== 31-Jan-2015::18:13:39 ===
Statistics database started.
 completed with 6 plugins.

=INFO REPORT==== 31-Jan-2015::18:13:39 ===
Server startup complete; 6 plugins started.
 * rabbitmq_management
 * rabbitmq_web_dispatch
 * webmachine
 * mochiweb
 * rabbitmq_management_agent
 * amqp_client

=ERROR REPORT==== 31-Jan-2015::18:13:40 ===
** Generic server rabbit_mgmt_external_stats terminating 
** Last message in was emit_update
** When Server state == {state,1048576}
** Reason for termination == 
** {badarg,[{erlang,list_to_binary,[undefined],[]},
            {rabbit_mgmt_external_stats,log_location,1,[]},
            {rabbit_mgmt_external_stats,'-infos/2-lc$^0/1-0-',2,[]},
            {rabbit_mgmt_external_stats,'-infos/2-lc$^0/1-0-',2,[]},
            {rabbit_mgmt_external_stats,emit_update,1,[]},
            {rabbit_mgmt_external_stats,handle_info,2,[]},
            {gen_server,handle_msg,5,[{file,"gen_server.erl"},{line,607}]},
            {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,227}]}]}

=INFO REPORT==== 31-Jan-2015::18:13:50 ===
    application: rabbitmq_management_agent
    exited: shutdown
    type: temporary
```

**Comment by vito-c** [comment]
I'm getting this issue with `3.4.2-management` but not with `3-management`

**Comment by yosifkit** [comment]
`3-management` is currently equal to `3.6.1-management`.  `3.4.2-management` did not contain the fixes (it is minimally `3.4.3` that had the fix from #7).
